docs/gen_ref_pages.py

```diff
@@ -0,0 +1,35 @@
+"""Generate the code reference pages."""
```

Comment:
Looks like it goes under `docs/mkdocs/reference`. 

I'm not sure this should go under a `.gitignore` either since this step is clearly not automated right now and is listed as a separate step before opening a PR so you have to commit the changes this script makes. 

https://docs.lineapy.org/0.2/guides/contributing/areas/update-docs/#updating-api-reference

Reply:
Ideally we move this command into our Makefile and include it as part of the documentation building / build process ... but we should do that later.

Reply:
Ok the documentation [here](https://mkdocstrings.github.io/recipes/) suggests we can leave it as a target in `gen-files` in `mkdocs.yml` and when I do this, I don't need the directory at all! I will still suggest keeping it there for now so we can see the diff.